### DISCUSSION TITLE
그냥 오브젝트와 하위 오브젝트의 차이에 따른 Transfom 변화 질문

### DISCUSSION THREAD

**Discussion by oh-bba-ya** [naverz/zepeto-studio-kor#1278] [5. Q&A]
토이 프로젝트 만들려고 이것저것 작성해보면서 해보는데
하위 오브젝트를 생성하게 된다면 Transform이 이상하게 적용되는것 같은데 이유를 모르겠네요 ㅠㅠ

다음 영상에서 처음엔 하위 오브젝트를 넣고 실행하면 큐브 모양이 이상해지고 하위 오브젝트가 아닌 상태로 넣으면 또 제대로 실행되더라고요 .. 이유를 모르겠네요 ㅠㅠ

https://user-images.githubusercontent.com/49023743/189673780-f2792127-8470-4e12-806b-9f3c488a2cca.mp4

오브젝트에 적용된 코드입니다!
var moveX = this.moveSpeed * Time.deltaTime;
        this.delta += Time.deltaTime;
        Debug.Log(this.delta);
        if (this.isRight) {

            if (this.delta >= this.RespawnTime) {
                Debug.Log("생성");
                Object.Instantiate(this.mObject);
                this.delta = 0;
            }

            this.transform.Translate(moveX, 0, 0);
            if (this.transform.position.x > 22) {
                Debug.Log("Remove");
                this.gameObject.SetActive(false);
            }
        }

**Comment by YSHFM** [comment]
유니티 하이어라키에서 하위 오브젝트의 Transform은 WorldSpace가 아닌 LocalSpace상으로 동작되는 것으로 알고 있습니다.
행렬변환값을 생각하시고 코드를 작성하시거나, transfrom의 localposition과 position값을 비교하시면서 적절히 사용하셔야 합니다.

행렬변환에 대해 자세히 아셔야 한다면 렌더링 파이프라인쪽 정보를 검색하시면 나오십니다.

**Reply by oh-bba-ya** [reply]
알려주셔서 감사합니다!

**Comment by zlmrzlo** [comment]
Scale은 웬만하면 111로 하시는 것이 좋을 것 같아요. Transform으로 원하는 결과가 나오지 않았을 때에는 Scale 문제일 수도 있을 것 같아요. 지금 영상만 봐도 Scale이 일정하지 않은 상태네요.

**Reply by oh-bba-ya** [reply]
알려주셔서 감사합니다!
